block/manager.go

```diff
@@ -495,11 +496,17 @@ func (m *Manager) publishBlock(ctx context.Context) error {
 
 		// set the validator set using the signer's public key
 		// TODO(ganesh): need to hook into a module that selects signers
-		pubKey, err := m.proposerKey.GetPublic().Raw()
+		pubKeyRaw, err := m.proposerKey.GetPublic().Raw()
 		if err != nil {
 			return err
 		}
-		block.SignedHeader.Validators = types.ValidatorSet{Validators: []types.Validator{{PublicKey: pubKey}}}
+		pubKey := ed25519.PubKey(pubKeyRaw)
+		proposer := &tmtypes.Validator{Address: pubKey.Address(), PubKey: pubKey}
+		// TODO: read staking query to construct validators
+		block.SignedHeader.Validators = &tmtypes.ValidatorSet{
+			Validators: []*tmtypes.Validator{proposer},
```

Comment:
i don't think you should assign the proposer to validator set. instead pick from the state current validators.

Reply:
you could use `m.lastState.Validators`

Reply:
Will handle this in #812 as it's blocking merge with lot of failing tests.